cmd/globalAlignmentAnchor/globalAlignmentAnchor.go

```diff
@@ -25,40 +25,78 @@ func writeToFileHandle(file io.Writer, species1 bed.Bed, species2 bed.Bed, score
+	pass := true
+
+	// chrom should be the same between species1 and species2 (special case: panTro6 chr2A, chr2B both count as a match to hg38 chr2)
+	pT6_hg38_chr2 := ((assembly_species1 == "hg38" && assembly_species2 == "panTro6" && chrom_species1 == "chr2" && ((chrom_species2 == "chr2A") || (chrom_species2 == "chr2B"))) || (assembly_species1 == "panTro6" && assembly_species2 == "hg38" && ((chrom_species1 == "chr2A") || (chrom_species1 == "chr2B")) && chrom_species2 == "chr2")) // the special case written as a bool variable
+	if chrom_species2 != chrom_species1 || pT6_hg38_chr2 {
+		pass = false
+	}
+
+	// maf entry should be roughly diagonal
+	if (float64(species2_ChromStart) <= float64(species1_ChromStart)-0.05*float64(species1_SrcSize)) || (float64(species2_ChromStart) >= float64(species1_ChromStart)+0.05*float64(species1_SrcSize)) {
+		pass = false
+	} else if (float64(species1_ChromStart) <= float64(species2_ChromStart)-0.05*float64(species2_SrcSize)) || (float64(species1_ChromStart) >= float64(species2_ChromStart)+0.05*float64(species2_SrcSize)) {
+		pass = false
+	}
+
+	return pass
+}
+
 // helper function: check if gap bed entry passes checks
-func gapBedPass(species1_ChromStart int, species1_ChromEnd int, species2_ChromStart int, species2_ChromEnd int, gapSizeProductLimit int) (bool, string, string) {
+// need both pos_species (most recently updated alignment position)
+// and species1_ChromStart (the start of the current bed entry)
+func gapBedPass(pos_species1 int, species1_ChromStart int, species1_ChromEnd int, pos_species2 int, species2_ChromStart int, species2_ChromEnd int, gapSizeProductLimit int) (bool, string, string) {
 	pass := true
 	species1_Name := "species1_gap"
 	species2_Name := "species2_gap"
+	// calculate gap size of current bed entry, based on species1_ChromStart
 	species1_gapSize := species1_ChromEnd - species1_ChromStart
 	species2_gapSize := species2_ChromEnd - species2_ChromStart
-	gapSizeMultiple := 0.0
-	gapSizeMultipleLimit := 100.00 //gapSizeMultipleLimit is currently hardcoded tentatively, 100
+	// calculate gap size since the most recently updated alignment position, based on pos_species
+	species1_gapSizeBig := species1_ChromEnd - pos_species1
+	species2_gapSizeBig := species2_ChromEnd - pos_species2
+	// define limits and necessary calculations to evaluate limit
+	// gap size product limit should be evaluted based on gapSize
 	gapSizeProduct := species1_gapSize * species2_gapSize
-
-	if species1_gapSize != 0 {
-		gapSizeMultiple = float64(species2_gapSize / species1_gapSize)
+	// gap size multiple should be evaluated based on gapSizeBig
+	gapSizeBigMultiple := 0.0
+	if species1_gapSizeBig != 0 {
+		gapSizeBigMultiple = float64(species2_gapSizeBig / species1_gapSizeBig)
 	}
+	gapSizeBigMultipleLimit := 100.00 // gapSizeBigMultipleLimit is currently hardcoded tentatively, 100
```

Comment:
Is this variable saying that when we have two sequences aligned to each other and one of them is 100 times longer than the other, we will consider the longer one to be an insertion?

Reply:
No, it is saying that if the currently aligned sequence in species2 is really far away from the previously aligned sequence in species2, 100 times further than the currently aligned sequence in species1 is from the previously aligned sequence in species1, then the current alignment should be discarded and not "trusted" to be a true alignment, because the currently aligned sequence in species2 is probably a random region further downstream on the chromosome, which breaks synteny. This is only a concern for species2 because species1 is always progressing linearly (has synteny) and a big gap in species1 might just mean a big insertion in species1.

Reply:
![IMG_3653](https://user-images.githubusercontent.com/75260661/173404758-a91bde51-971d-45bc-ab49-1ec0e39a48bc.jpg)